src/main/kotlin/io/github/anki/anki/service/CardsService.kt

```diff
@@ -1,32 +1,74 @@
 package io.github.anki.anki.service
 
+import io.github.anki.anki.controller.exceptions.CardDoesNotExistException
 import io.github.anki.anki.repository.mongodb.CardRepository
+import io.github.anki.anki.repository.mongodb.document.MongoCard
 import io.github.anki.anki.service.model.Card
 import io.github.anki.anki.service.model.mapper.toCard
 import io.github.anki.anki.service.model.mapper.toMongo
+import org.bson.types.ObjectId
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 
 @Service
 class CardsService(
     private val cardRepository: CardRepository,
+    private val deckService: DeckService,
 ) {
-
-    fun createNewCard(card: Card): Card {
+    fun createNewCard(userId: String, card: Card): Card {
+        deckService.getDeckByIdAndUserId(card.deckId, userId)
         LOG.info("Creating new card: {}", card)
-        return cardRepository.insert(
-            card.toMongo(),
-        )
+        return cardRepository.insert(card.toMongo())
             .toCard()
             .also { LOG.info("Successfully saved new card: {}", it) }
     }
 
-    fun deleteCard(cardId: String) {
+    fun getAllCardsFromDeck(deckId: String, userId: String): List<Card> {
+        deckService.getDeckByIdAndUserId(deckId, userId)
+        LOG.info("Getting MongoCards by deckId = {}", deckId)
+        return cardRepository.findByDeckId(ObjectId(deckId))
+            .map { it.toCard() }
+            .also { LOG.info("Got MongoCards from DB {}", it) }
+    }
+
+    fun updateCard(userId: String, card: Card): Card {
+        deckService.getDeckByIdAndUserId(card.deckId, userId)
```

Comment:
Don't think that you need to ensure that deck is exists

Reply:
thats how i check that user owns this deck
if deck was not found i will throw the exception and client will see "Deck does not exist"

Reply:
For this case I suggest refactoring)

Looks much more logically
```kotlin
        if (!deckService.hasAccess(userId, card.deckId)) {
            throw IllegalArgumentException("User does not have access to the deck")
        }
```